conans/client/generators/cmake_find_package.py

```diff
@@ -74,8 +74,8 @@ class CMakeFindPackageGenerator(GeneratorComponentsMixin, Generator):
 
         set({{ pkg_name }}_COMPONENTS {{ pkg_components }})
 
-        if({{ pkg_name }}_FIND_COMPONENTS)
-            foreach(_FIND_COMPONENT {{ '${'+pkg_name+'_FIND_COMPONENTS}' }})
+        if({{ pkg_filename }}_FIND_COMPONENTS)
```

Comment:
was this a bug? How is this connected to not listening to properties?

Reply:
Yes, I commented this here, as [it's also happening for CMakeDeps](https://github.com/conan-io/conan/pull/10077/files#r756964886), I have extended  `test_component_not_found_cmake` to cover this.
The thing is that CMake generates a `<PackageName>_FIND_COMPONENTS` when `COMPONENTS` are specified in the `find_package`, but we are using `pkg_name` here that may not be the same as the `<PackageName>` CMake variable that is equivalent to our `filename`.

Reply:
I was going the fix this in the PR for CMakeDeps and absolute targets but I can fix all of them in another PR as well...